examples/player/simid_player.js

```diff
@@ -609,14 +624,27 @@ class SimidPlayer {
     console.log('The creative has asked for the player to ping ' + requestedUrlArray);
   }
 
+  /**
+   * Called when creative requests a change in duration of ad.
+   * @private
+   */
   onRequestChangeAdDuration(incomingMessage) {
-    if (this.requestedDuration_ != NO_REQUESTED_DURATION) {
-      // TODO: Support multiple change duration requests.
-      this.simidProtocol.reject(incomingMessage);
+    this.requestedDuration_  = incomingMessage.args['duration'];
+    //If requested duration is negative seconds, play entire ad
+    if (this.requestedDuration_ < 0) {
+      this.sendLog("requested duration too short: " + this.requestedDuration_ + " seconds");
+      setTimeout(() => {
+        this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+      }, this.adVideoElement_.duration * 1000);
+    } //If requested duration is 0 seconds, stop ad immediately  
+    else if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+      this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
```

Comment:
you need to always resolve or reject the requestChangeDuration message.

Reply:
fixed